sdk/src/invariant.ts

```diff
@@ -53,72 +53,87 @@ export class Invariant {
     network: Network,
+    abi: any,
+    address: string,
     storageDepositLimit: number | null = null,
     refTime: number = DEFAULT_REF_TIME,
-    proofSize: number = DEFAULT_PROOF_SIZE,
-    abi: any,
-    deploymentAddress: string
+    proofSize: number = DEFAULT_PROOF_SIZE
   ) {
     this.api = api
+    this.waitForFinalization = network !== Network.Local
+    this.contract = new ContractPromise(this.api, abi, address)
     this.gasLimit = api.registry.createType('WeightV2', {
       refTime,
       proofSize
     }) as WeightV2
     this.storageDepositLimit = storageDepositLimit
-    this.waitForFinalization = network != Network.Local
-    this.contract = new ContractPromise(this.api, abi, deploymentAddress)
     this.abi = new Abi(abi)
   }
 
   static async getContract(
     api: ApiPromise,
-    account: IKeyringPair,
-    storageDepositLimit: number | null = null,
-    refTime: number = DEFAULT_REF_TIME,
-    proofSize: number = DEFAULT_PROOF_SIZE,
-    initFee: Percentage,
-    network: Network
+    network: Network,
+    account?: IKeyringPair,
+    fee?: Percentage,
+    address?: string,
+    options?: ContractOptions
   ): Promise<Invariant> {
-    const invariantData = await getDeploymentData('invariant')
-    if (process.env.INVARIANT_ADDRESS && network != Network.Local) {
-      return new Invariant(
-        api,
-        network,
-        storageDepositLimit,
-        refTime,
-        proofSize,
-        invariantData.abi,
-        process.env.INVARIANT_ADDRESS
-      )
-    } else {
-      const invariantDeploy = await Invariant.deploy(
-        api,
-        account,
-        invariantData.abi,
-        invariantData.wasm,
-        initFee
-      )
-      const invariant = new Invariant(
-        api,
-        Network.Local,
-        storageDepositLimit,
-        refTime,
-        proofSize,
-        invariantData.abi,
-        invariantDeploy.address
-      )
-      return invariant
+    if (address) {
+      return Invariant.load(api, network, address, options)
+    }
+
+    if (account) {
+      return Invariant.deploy(api, network, account, fee, options)
     }
+
+    throw new Error('Invalid arguments')
   }
 
   static async deploy(
     api: ApiPromise,
+    network: Network,
     account: IKeyringPair,
-    abi: any,
-    wasm: Buffer,
-    fee: Percentage
-  ): Promise<DeployedContract> {
-    return deployContract(api, account, abi, wasm, 'new', [fee])
+    fee: Percentage = { v: 0n },
+    options?: ContractOptions
+  ): Promise<Invariant> {
+    const deploymentData = await getDeploymentData('invariant')
+    const deploy = await deployContract(
+      api,
+      account,
+      deploymentData.abi,
+      deploymentData.wasm,
+      'new',
+      [fee]
+    )
+
+    return new Invariant(
+      api,
+      network,
+      deploymentData.abi,
+      deploy.address,
+      options?.storageDepositLimit,
+      options?.refTime,
+      options?.proofSize
+    )
+  }
+
+  static async load(
```

Comment:
Mark all functions that you assume should not be directly called by the sdk client as `private`. It appears that `load` is one such function